builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -38,6 +39,20 @@ func resourceAwsSecurityGroup() *schema.Resource {
 				},
 			},
 
+			"name_prefix": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters, name is limited to 255", k))
```

Comment:
I think this error message is confusing, since we don't specify what 255 actually is. Some people will figure it out, but I don't think it adds anything. The API says the limit is 255 characters as well, so I'm not sure why we error if the length is over 100 (unless I'm misunderstanding something about Go's `len` function)